Geometry_Engine/Query/HashArray.cs

```diff
@@ -45,13 +45,33 @@ public static partial class Query
             "\nThe number of points is reduced to the minimum essential to determine uniquely any geometry." +
             "\nAdditionally, the resulting points are transformed based on the source geometry type, to remove or minimize collisions." +
             "\n(Any transformation so performed is translational only, in order to support geometrical tolerance, i.e. numerical distance, when comparing GeometryHashes downstream).")]
+        [Input("igeometry", "Geometry you want to compute the hash array for.")]
+        [Input("comparisonConfig", "Configurations on how the hash array is computed, with options for numerical approximation, type exceptions and many others.")]
         [Output("hashArray", "Array of numbers representing a unique signature of the input geometry.")]
-        public static double[] IHashArray(this IGeometry igeometry)
+        public static double[] IHashArray(this IGeometry igeometry, BaseComparisonConfig comparisonConfig = null)
+        {
+            return HashArray(igeometry as dynamic, 0, comparisonConfig, null);
+        }
+
+        /***************************************************/
+
+        [Description("Returns a signature of the input geometry, useful for distance-based comparisons and diffing." +
+            "\nThe hash is computed as an array representing the coordinate of significant points taken on the geometry." +
+            "\nThe number of points is reduced to the minimum essential to determine uniquely any geometry." +
+            "\nAdditionally, the resulting points are transformed based on the source geometry type, to remove or minimize collisions." +
+            "\n(Any transformation so performed is translational only, in order to support geometrical tolerance, i.e. numerical distance, when comparing GeometryHashes downstream).")]
+        [Input("igeometry", "Geometry you want to compute the hash array for.")]
+        [Input("comparisonConfig", "Configurations on how the hash array is computed, with options for numerical approximation, type exceptions and many others.")]
+        [Input("fullName", "Name of the property that holds the target object to calculate the hash array for. This name will be used to seek any matching custom configuration to apply against the `comparisonConfig` input.")]
+        [Output("hashArray", "Array of numbers representing a unique signature of the input geometry.")]
+        public static double[] IHashArray(this IGeometry igeometry, BaseComparisonConfig comparisonConfig, string fullName = null)
         {
             if (igeometry == null)
                 return new double[] { };
 
-            return HashArray(igeometry as dynamic, 0);
+            fullName = fullName ?? igeometry.GetType().FullName;
```

Comment:
Because if `fullName` is null, then the input object is not considered a property of some other object, but a stand alone object, whose name is its FullName.